src/sentry/api/base.py

```diff
@@ -213,14 +218,28 @@ def permission_denied(self, request, message=None, code=None):
         the appropriate exception according to parent DRF function.
         """
         permissions = self.get_permissions()
+        if request.user.is_authenticated and len(permissions) == 1:
+            permission_cls = permissions[0]
+            use_staff = features.has("auth:enterprise-staff-cookie", actor=request.user)
+
+            # TODO(schew2381): Remove SuperuserOrStaffFeatureFlaggedPermission
+            # from isinstance checks once feature flag is removed.
+            can_be_superuser = request.user.is_superuser
+            has_only_superuser_permission = isinstance(
+                permission_cls, (SuperuserPermission, SuperuserOrStaffFeatureFlaggedPermission)
+            )
 
-        can_be_superuser = request.user.is_authenticated and request.user.is_superuser
-        has_only_superuser_permission = len(permissions) == 1 and isinstance(
-            permissions[0], SuperuserPermission
-        )
+            if not use_staff and can_be_superuser and has_only_superuser_permission:
+                raise SuperuserRequired()
+
+            can_be_staff = request.user.is_authenticated and request.user.is_staff
+            has_only_staff_permission = isinstance(
+                permission_cls, (StaffPermission, SuperuserOrStaffFeatureFlaggedPermission)
+            )
+
+            if use_staff and can_be_staff and has_only_staff_permission:
+                raise StaffRequired()
```

Comment:
```suggestion
            if not use_staff:
                if can_be_superuser and has_only_superuser_permission:
                    raise SuperuserRequired()
            else:
               # enforce staff permission if feature flag is on
                can_be_staff = request.user.is_authenticated and request.user.is_staff
                has_only_staff_permission = isinstance(
                    permission_cls, (StaffPermission, SuperuserOrStaffFeatureFlaggedPermission)
                )
                if can_be_staff and has_only_staff_permission:
                    raise StaffRequired()
```

could we group the blocks into something like this so that we can easily tell what happens with or without the feature flag?

Reply:
I actually think the first one is more readable imo, b/c the bool conditions read like english.
I think it's better to avoid the `else` so you don't have to remember the `use_staff` condition being true as you're reading the check for raising `StaffRequired`

Reply:
you could switch the blocks around too. i was grouping your flow into the if/else. i agree it's more readable with `if use_staff` first